src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,41 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type == Bulldozer.class) {
+            return new ArrayList<>(new BulldozerProducer().get());
```

Comment:
```suggestion
        if (type == Bulldozer.class) {
            return new ArrayList<>(bulldozerProducer.get());
```
better create class variables to avoid creating a new producer each time this method is invoked
remember to use abstract reference type